scipy/spatial/transform/tests/test_rigid_transform.py

```diff
@@ -29,824 +36,1151 @@ def test_repr():
+    pytest.importorskip("jax")
+    import jax
+    import jax.numpy as jp
+
+    t = jp.asarray([1, 2, 3])
+    from_translation = jax.jit(RigidTransform.from_translation)
+    jax.block_until_ready(from_translation(t))
+
+
+def test_from_matrix(xp):
     atol = 1e-12
 
     # Test single transform matrix
-    matrix = np.eye(4)
-    matrix[:3, 3] = [1, 2, 3]
+    matrix = xp.eye(4)
+    matrix = xpx.at(matrix)[..., :3, 3].set(xp.asarray([1, 2, 3]))
     tf = RigidTransform.from_matrix(matrix)
-    assert_allclose(tf.as_matrix(), matrix, atol=atol)
+    xp_assert_close(tf.as_matrix(), matrix, atol=atol)
     assert tf.single
 
     # Test multiple transform matrices
-    matrices = np.array([np.eye(4)]*2)
-    matrices[0, :3, 3] = [1, 2, 3]
-    matrices[1, :3, 3] = [4, 5, 6]
+    # torch compat had issues with repeat, so we avoid using it here for testing. See
+    # https://github.com/data-apis/array-api-compat/issues/292
+    matrices = xp.zeros((2, 4, 4))
+    matrices = xpx.at(matrices)[..., :4, :4].set(xp.eye(4))
+    matrices = xpx.at(matrices)[..., :3, 3].set(xp.asarray([[1, 2, 3], [4, 5, 6]]))
     tf = RigidTransform.from_matrix(matrices)
-    assert_allclose(tf.as_matrix(), matrices, atol=atol)
+    xp_assert_close(tf.as_matrix(), matrices, atol=atol)
     assert not tf.single
 
     # Test non-1 determinant
-    matrix = np.diag([2, 2, 2, 1])
+    matrix = xp.eye(4)
+    matrix = xpx.at(matrix)[..., :3, :3].set(xp.eye(3) * 2.0)
     tf = RigidTransform.from_matrix(matrix)
-    assert_allclose(tf.as_matrix(), np.eye(4), atol=atol)
+    xp_assert_close(tf.as_matrix(), xp.eye(4), atol=atol)
 
     # Test non-orthogonal rotation matrix
-    matrix = np.array([[1, 1, 0, 0],
-                       [0, 1, 0, 0],
-                       [0, 0, 1, 0],
-                       [0, 0, 0, 1]])
+    matrix = xp.asarray([[1, 1, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]])
     tf = RigidTransform.from_matrix(matrix)
-    expected = np.array([[0.894427,  0.447214, 0, 0],
-                         [-0.447214,  0.894427, 0, 0],
-                         [0, 0, 1, 0],
-                         [0, 0, 0, 1]])
-    assert_allclose(tf.as_matrix(), expected, atol=1e-6)
+    expected = xp.asarray(
+        [
+            [0.894427, 0.447214, 0.0, 0.0],
+            [-0.447214, 0.894427, 0.0, 0.0],
+            [0.0, 0.0, 1.0, 0.0],
+            [0.0, 0.0, 0.0, 1.0],
+        ]
+    )
+    xp_assert_close(tf.as_matrix(), expected, atol=1e-6)
 
     # Test invalid matrix
-    with pytest.raises(ValueError):
-        invalid = np.eye(4)
-        invalid[3, 3] = 2  # Invalid last row
-        RigidTransform.from_matrix(invalid)
+    invalid = xp.eye(4)
+    invalid = xpx.at(invalid)[..., 3, 3].set(2)  # Invalid last row
+    if is_lazy_array(invalid):
+        tf = RigidTransform.from_matrix(invalid)
+        assert xp.all(xp.isnan(tf.as_matrix()))
+    else:
+        with pytest.raises(ValueError):
+            RigidTransform.from_matrix(invalid)
```

Comment:
Do users of lazy backends have the means to validate their input? e.g. is there an eager user-facing validation function like `scipy.cluster.is_valid_linkage`?